### PR TITLE
uI/15399747 main navigation

### PR DESCRIPTION
**Pull request by KatarzynaStefanowiczPagepro**
## Description

- [] TODO write description

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

### CODE REVIEW

**Comment by KatarzynaStefanowiczPagepro** [comment]
https://tw.auroracreation.com/app/tasks/15399747

Main navigation with the dropdown on hover.
Added sub-navigation on mobile.

image

image

File: src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                )}
+              </span>
+              <p className="text-gray-600 text-md">
+                {formatMessage({
+                  id: `subMenuItem.${label}`,
+                  defaultMessage: `${description}`,
+                })}
+              </p>
+            </a>
+            {/* </Link> */}
```
Comment (CzarekDryl) [inline src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx:42]: Is it needed?

File: src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const isMobile = useMobile();
+  const { formatMessage } = useIntl();
+  const Wrapper = isMobile ? 'div' : Card;
+
+  return (
+    <Wrapper hasShadow={!isMobile}>
+      <ul className={clsx(styles.subMenuList, 'sm:grid-cols-[repeat(2,1fr)] md:grid-cols-[repeat(3,1fr)]')}>
+        {items.map(({ label, href, description, status }) => (
+          <li key={label}>
+            {/* <Link to="/"> */}
```
Comment (CzarekDryl) [inline src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx:25]: Is it needed?

File: src/components/common/Extensions/MainNavigation/partials/Nav/Nav.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      {item.subMenu && (
+        <div className={styles.subMenuMobile}>
+          <AnimatePresence>
+            {isOpen && (
+              <motion.div
+                key="accordion-content"
+                initial={{ opacity: 0, height: 0 }}
+                animate={{ opacity: 1, height: 'auto' }}
+                exit={{ opacity: 0, height: 0 }}
+                transition={{ duration: 0.4, ease: 'easeOut' }}
```
Comment (CzarekDryl) [inline src/components/common/Extensions/MainNavigation/partials/Nav/Nav.tsx:62]: I think it's better to use variants here image

File: src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx (hunk 1 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx:42]: Yes, we will have to use Routing here.

File: src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx (hunk 2 above)
Comment (KatarzynaStefanowiczPagepro) [reply src/components/common/Extensions/MainNavigation/partials/SubMenu/SubMenu.tsx:25]: Yes, we will have to use Routing here.

**Review by CzarekDryl — APPROVED** [review]

File: src/components/common/Extensions/MainNavigation/partials/Nav/Nav.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    <a className={styles.navLink} href={item.href}>
+      {formatMessage({
+        id: `mainNavItem.${item.label}`,
+        defaultMessage: `${item.label}`,
+      })}
+    </a>
+  );
+
+  return isMobile ? (
+    <>
```
Comment (krawczykkatarzyna) [inline src/components/common/Extensions/MainNavigation/partials/Nav/Nav.tsx:38]: please move mobile view to another one file

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**